src/D2L.Bmx/Program.cs

```diff
@@ -210,25 +210,30 @@
 // start bmx
 return await new CommandLineBuilder( rootCommand )
 	.UseDefaults()
-	.AddMiddleware( async (
-		context,
-		next
-	) => {
-		await UpdateChecker.CheckForUpdatesAsync( configProvider.GetConfiguration() );
-		//Creating new Cache directory and removing old files.
-		if( !Directory.Exists( BmxPaths.CACHE_DIR ) ) {
-			try {
-				Directory.CreateDirectory( BmxPaths.CACHE_DIR );
-				File.Delete( Path.Join( BmxPaths.BMX_DIR, "awsCredsCache" ) );
-				File.Delete( Path.Join( BmxPaths.BMX_DIR, "sessions" ) );
-				File.Delete( Path.Join( BmxPaths.BMX_DIR, "update_check" ) );
-			} catch( Exception ex ) {
-				Console.Error.WriteLine( "Error cleaning up old files continuing..." );
+	.AddMiddleware(
+		middleware: async ( context, next ) => {
+			// initialize BMX directories
+			if( !Directory.Exists( BmxPaths.CACHE_DIR ) ) {
+				try {
+					// the cache directory is inside the config directory (~/.bmx), so this ensures both exist
+					Directory.CreateDirectory( BmxPaths.CACHE_DIR );
+					// move the old sessions file (v3.0-) into the new cache directory (v3.1+)
+					if( File.Exists( BmxPaths.SESSIONS_FILE_LEGACY_NAME ) ) {
+						File.Move( BmxPaths.SESSIONS_FILE_LEGACY_NAME, BmxPaths.SESSIONS_FILE_NAME );
+					}
+				} catch( Exception ex ) {
+					throw new BmxException( "Failed to initialize BMX directory (~/.bmx)", ex );
```

Comment:
Don't swallow errors on directory init, because our file logic in the rest of the codebase now assumes the necessary directories always exist. If there's an error here and is ignored, we may encounter an unexpected error down the line, which results in non-descriptive error message for the user.

This does mean that we run the slight risk of erroring out on failing to move the sessions file, when it could be ignored. But that's a much smaller concern to me. If there's an error in moving that file, there's probably something odd about the user's filesystem and may be worth investigating anyway.